website/src/pages/downloads/ProjectCard.js

```diff
@@ -0,0 +1,286 @@
+import IconHexagon from "../../assets/icons/hexagon.svg";
+
+const Dropdown = (props) => {
+  const ref = useRef();
+  const { isDropdownOpen, setIsDropdownOpen } = props;
+  useOutsideClick(ref, () => {
+    if (isDropdownOpen) {
+      setIsDropdownOpen(false);
+    }
+  });
+  return (
+    <StyledDropdown ref={ref} open={isDropdownOpen}>
+      {props.children}
+    </StyledDropdown>
+  );
+};
+
+const ProjectCard = (props) => {
+  const [isDropdownOpen, setIsDropdownOpen] = useState(false);
+  const [repoStats, setRepoStats] = useState({ stars: 0, issues: 0 });
+  const {
+    name,
+    description,
+    shape,
+    color,
+    version,
+    releaseDate,
+    githubRepo,
+  } = props;
+  const shapeComponent =
+    shape === "triangle" ? (
+      <IconTriangle />
+    ) : shape === "square" ? (
+      <IconSquare />
+    ) : (
+      <IconHexagon />
+    );
+  const downloadLink = `https://www.apache.org/dyn/closer.cgi/apisix${
+    "/" + paramCase(name.replace("APISIX™", ""))
+  }/${version}/apache-${paramCase(name.replace("™", ""))}-${version}-src`;
+
+  useEffect(() => {
+    getGitHubRepoStats(githubRepo).then((stats) => {
+      setRepoStats({
+        stars: stats.stargazers_count,
+        issues: stats.open_issues_count,
+      });
+    });
+  }, []);
+
+  return (
+    <Card>
+      <LeftSide>
+        <Title>
+          <ShapeBeforeTitle color={color}>{shapeComponent}</ShapeBeforeTitle>
+          {name}
+        </Title>
+        <Description>{description}</Description>
+        <LeftSideLinks>
+          <LeftSideLink
+            href={`https://github.com/${githubRepo}`}
+            target="_blank"
+            title="Stars"
+          >
+            <IconStar /> {repoStats.stars}
+          </LeftSideLink>
+          <LeftSideLink
+            href={`https://github.com/${githubRepo}/issues`}
+            target="_blank"
+            title="Issues"
+          >
+            <IconInfo /> {repoStats.issues}
+          </LeftSideLink>
+          <LeftSideLink
+            href={`https://github.com/${githubRepo}/blob/master/CHANGELOG.md`}
+            target="_blank"
+          >
+            <IconDocumentText /> CHANGELOG
+          </LeftSideLink>
+        </LeftSideLinks>
+      </LeftSide>
+      <RightSide>
+        <VersionInfo>
+          Latest Version · <span>{version}</span>
+          <br />
+          Release Date · <span>{releaseDate}</span>
+        </VersionInfo>
+        <div>
+          <Button
+            onClick={() => setIsDropdownOpen(!isDropdownOpen)}
+            background={color}
+          >
+            <IconDownload /> Download
+          </Button>
+          <Dropdown
+            isDropdownOpen={isDropdownOpen}
+            setIsDropdownOpen={setIsDropdownOpen}
+          >
+            <DropdownItem href={`${downloadLink}.tgz`} target="_blank">
+              Source
+            </DropdownItem>
+            <DropdownItem href={`${downloadLink}.tgz.asc`} target="_blank">
+              ASC
```

Comment:
Hi @qier222, only the Source link needs a mirror, both ASC and SHA512 should use link like `https://downloads.apache.org/apisix/dashboard/2.4/apache-apisix-dashboard-2.4-src.tgz.asc`

Reply:
Please recheck those links from `https://apisix.apache.org/downloads`